File: esap-core/src/main/java/ru/javavlsu/kb/esap/repository/UserDeviceTokenRepository.java
package ru.javavlsu.kb.esap.repository;

import org.springframework.data.jpa.repository.JpaRepository;
import ru.javavlsu.kb.esap.model.User;
import ru.javavlsu.kb.esap.model.UserDeviceToken;

import java.util.List;
import java.util.Optional;

public interface UserDeviceTokenRepository extends JpaRepository<UserDeviceToken, Long> {

    List<UserDeviceToken> getUserDeviceTokensByUser(User user);

    boolean existsByTokenAndUser(String token, User user);

    Optional<UserDeviceToken> findUserDeviceTokenByToken(String token);
}

File: notification-service/src/main/kotlin/ru/javavlsu/kb/notificationservice/repository/UserDeviceTokenRepository.kt
package ru.javavlsu.kb.notificationservice.repository

import org.springframework.data.mongodb.repository.MongoRepository
import org.springframework.stereotype.Repository
import ru.javavlsu.kb.notificationservice.model.UserDeviceToken
import java.util.*

/**
 * UserDeviceTokenRepository 23.07.2026 thewyolar
 * Copyright (c) 2026.
 */
@Repository
interface UserDeviceTokenRepository : MongoRepository<UserDeviceToken, Long> {

    fun getUserDeviceTokensByUserId(userId: Long): List<UserDeviceToken>

    fun existsByTokenAndUserId(token: String, userId: Long): Boolean

    fun findUserDeviceTokenByToken(token: String): Optional<UserDeviceToken>
}
